### ISSUE TITLE
Truncation error on Node backend

### ISSUE THREAD

**Issue by stefan-hoeck** [idris-lang/Idris2#1037]
# Steps to Reproduce
Run

```idris
main : IO ()
main = do printLn (map (*0) [the Bits8 0])
          printLn [the Bits8 0]
          printLn (map (*0) [the Bits8 0] == [0])
```

# Expected Behavior
Prints
```
[0]
[0]
True
```

# Observed Behavior
Prints
```
[256]
[0]
True
```

Node shows similar behavior with other bounded types, printing the theoretically unreachable upper bound in each case.
I'm trying to untangle the generated JS code but am having a hard time figuring out what's going on here.

**Comment by stefan-hoeck** [comment]
Seems like there's an error in `boundedUInt` in `src/Compiler/ES/ES.idr`. Will try and submit a fix.

**Comment by stefan-hoeck** [comment]
Should be fixed by #1038.
